Added Receive + Prometheus Agent tutorial. (#4817)
thanos-io/thanos: tutorials/katacoda/thanos/4-receiver-agent/intro.md

# Intermediate: (Bonus) Using Prometheus Agent for pure forward-only metric streaming with Thanos Receive

The [Thanos](thanos.io) project defines a set of components that can be composed together into a highly available metric system with **unlimited storage capacity** that **seamlessly** integrates into your existing Prometheus deployments.

But [Promoetheus](https://prometheus.io/) project is far from stagnation. Together with the community, it constantly evolves to bring value to different network and cluster topologies that we see. Thanos brings distributed and cloud storage and querying to the table, built on the core Prometheus code and design. It allowed Prometheus to focus on critical collection and single cluster monitoring functionalities.

As we learned in the previous tutorial, certain situations require us to collect (pull) data from applications and stream them out of the cluster as soon as possible. `Thanos Receive` allows doing that by ingesting metrics using the Remote Write protocol that the sender can implement. Typically we recommended using Prometheus with short retention and blocked read and query API as a "lightweight" sender.

> From November 2021, however, we, Prometheus maintainers with the community, introduced a brand new Prometheus mode called "Agent mode". The agent mode is optimized for efficient metric scraping and forwarding (e., immediate metric removal once it's securely delivered to a remote location). Since this is incredibly helpful for the Thanos community, we wanted to give you first-hand experience deploying Prometheus Agent together with Thanos Receive in this course.

Reviewer: I'm not going to push too hard on this, but a light shoutout to Grafana Agent would be lovely :)

Author: Yes!

Author: Will be in blog post anyway, but happy to add here too

Author: Added, PTAL

Reviewer: Perfect, thanks!

# Intermediate: (Bonus) Using Prometheus Agent for pure forward-only metric streaming with Thanos Receive

The [Thanos](thanos.io) project defines a set of components that can be composed together into a highly available metric system with **unlimited storage capacity** that **seamlessly** integrates into your existing Prometheus deployments.

But [Prometheus](https://prometheus.io/) project is far from slowing down the development. Together with the community, it constantly evolves to bring value to different network and cluster topologies that we see. Thanos brings distributed and cloud storage and querying to the table, built on the core Prometheus code and design. It allowed Prometheus to focus on critical collection and single cluster monitoring functionalities.

As we learned in the previous tutorial, certain situations require us to collect (pull) data from applications and stream them out of the cluster as soon as possible. `Thanos Receive` allows doing that by ingesting metrics using the Remote Write protocol that the sender can implement. Typically we recommended using Prometheus with short retention and blocked read and query API as a "lightweight" sender.

> From November 2021, however, we, Prometheus maintainers with the community, introduced a brand new Prometheus mode called "Agent mode". The implementation itself was already battle tested on https://github.com/grafana/agent, where it was available and authored by [Robert Fratto](https://github.com/rfratto) since 2020.

Reviewer 2: Nit, maybe just a bit more simply:
```suggestion
In November 2021, however, we, the Prometheus community, introduced a brand new Prometheus mode called "Agent mode". The implementation itself was already battle tested on https://github.com/grafana/agent, where it was available and authored by [Robert Fratto](https://github.com/rfratto) since 2020.
```

> The agent mode is optimized for efficient metric scraping and forwarding (e., immediate metric removal once it's securely delivered to a remote location). Since this is incredibly helpful for the Thanos community, we wanted to give you first-hand experience deploying Prometheus Agent together with Thanos Receive in this course.

Reviewer 2: ```suggestion
The agent mode is optimized for efficient metric scraping and forwarding (i.e. immediate metric removal once it's securely delivered to a remote location). Since this is incredibly helpful for the Thanos community, we wanted to give you first-hand experience deploying Prometheus Agent together with Thanos Receive in this course.
```

In this tutorial, you will learn:

> * How to slim down Prometheus based client-side metric collection to a minimum, using the new Prometheus "Agent mode" with `Thanos Receiver`, explained in the previous tutorial.

Reviewer 3: ```suggestion
* How to reduce Prometheus based client-side metric collection to a minimum, using the new Prometheus "Agent mode" with `Thanos Receiver`, explained in the previous tutorial.
```

> * How to slim-down Prometheus based client-side metric collection to a minimum using new Prometheus Agent mode with Thanos Receiver explained in the previous tutorial.

Reviewer 2: Nit, I would keep the double quotes consistently in all places.

```suggestion
* How to slim down Prometheus based client-side metric collection to a minimum, using the new Prometheus "Agent mode" with `Thanos Receiver`, explained in the previous tutorial.
```
